Source/WebCore/html/parser/HTMLElementStack.h

```diff
@@ -48,16 +48,16 @@ class HTMLElementStack {
     class ElementRecord {
         WTF_MAKE_NONCOPYABLE(ElementRecord); WTF_MAKE_FAST_ALLOCATED;
     public:
-        ElementRecord(Ref<HTMLStackItem>&&, std::unique_ptr<ElementRecord>);
+        ElementRecord(HTMLStackItem&&, std::unique_ptr<ElementRecord>);
         ~ElementRecord();
-    
-        Element& element() const { return m_item->element(); }
-        ContainerNode& node() const { return m_item->node(); }
-        const AtomString& namespaceURI() const { return m_item->namespaceURI(); }
-        HTMLStackItem& stackItem() { return m_item.get(); }
-        const HTMLStackItem& stackItem() const { return m_item.get(); }
 
-        void replaceElement(Ref<HTMLStackItem>&&);
+        Element& element() const { return m_item.element(); }
```

Comment:
It is not new but it is a little peculiar that this function is const but returns a non-const reference.

Reply:
I started doing this, but then found it caused some problems so I am just going to leave it alone. There is an argument that the stack element points to the DOM element, so the reference can be non-const because of that. Just as a `const RefPtr<Element>` can yield an `Element*`, but the real argument is that trying to remove this makes a mess.